centrallix/osdrivers/objdrv_sybase.c

```diff
@@ -807,17 +808,20 @@ sybd_internal_GetCxValue(void* ptr, int ut, pObjData val, int datatype)
 		divtmp = msl/10000;
 		n = (n<<16) + divtmp;
 		msl -= divtmp*10000;
-		val->Money->WholePart = n;
-		val->Money->FractionPart = msl;
+		val->Money->Value = (n*10000) + msl;
 		if (minus)
-		    {
-		    val->Money->WholePart = -val->Money->WholePart;
-		    if (val->Money->FractionPart > 0)
-			{
-			val->Money->WholePart--;
-			val->Money->FractionPart = 10000 - val->Money->FractionPart;
-			}
-		    }
+		    val->Money->Value = -val->Money->Value;
```

Comment:
The original objdrv_sybase.c code here was written in a world where there was no int64 / long long, and the 8-byte value at **ptr** at line 786 in the original code (memcpy(&lsl, ptr+4, 4)) is an int64 (long long), most significant little endian int32 first and least significant little endian int32 second.  So it's mixed-endian.  But once the mixed-endianness is corrected, the value is correct as a long long and is in 10000ths of a currency unit, facilitating a direct assignment to ->Value.

Thus the original code had to divide an int64 by 10000 without int64 support.  That is why it is complex, and does long division one 16-bit word at a time.  The code can be vastly simplified now. :)

Reply:
Is the mixed-endianness corrected in the existing code during the long division section?  Or do I need to write new code for that fix before assigning to Value?

Reply:
The existing code deals with it via the msl and lsl values - which are LE 32 bit values.  In your case, when you copy the buffer value into the long long, recognize that you'll need to swap the two 32 bit halves.

I realize this can't be tested without a SAP Sybase ASE server, and we don't have one set up on our servers for testing right now.  I'll do the testing of this on a client's staging instance.  SAP Sybase ASE for Linux is free, so it'd be good for us to get one set up at some point.